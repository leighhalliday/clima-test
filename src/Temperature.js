import React from "react";
import PropTypes from "prop-types";

export default class Temperature extends React.Component {
  static propTypes = {
    temp: PropTypes.number.isRequired,
    city: PropTypes.string.isRequired,
    toggleForecast: PropTypes.func.isRequired
  };

  render() {
    return (
      <div
        onClick={e => {
          e.preventDefault();
          this.props.toggleForecast();
        }}
      >
        <h2>{this.props.temp}&deg;c</h2>
        <h3>{this.props.city}</h3>
      </div>
    );
  }
}
