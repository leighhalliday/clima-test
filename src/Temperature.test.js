/* global React, render, mount, sinon */
import Temperature from "./Temperature";

it("renders component", () => {
  const wrapper = render(
    <Temperature temp={10} city="Toronto" toggleForecast={() => {}} />
  );
  expect(wrapper).toMatchSnapshot();
});

it("formats temp correctly", () => {
  const wrapper = mount(
    <Temperature temp={10} city="Toronto" toggleForecast={() => {}} />
  );

  const text = wrapper.find("h2").text();

  expect(text).toEqual("10Â°c");
});

it("calls toggleForecast on click", () => {
  const spy = sinon.spy();
  const wrapper = mount(
    <Temperature temp={10} city="Toronto" toggleForecast={spy} />
  );

  wrapper
    .find("div")
    .first()
    .simulate("click");

  expect(spy.calledOnce).toBe(true);
});
