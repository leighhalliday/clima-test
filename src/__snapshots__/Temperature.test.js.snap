// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component 1`] = `
<div>
  <h2>
    10Â°c
  </h2>
  <h3>
    Toronto
  </h3>
</div>
`;
